### PR TITLE
[Okta v2] Make API Token Non-required When Using OAuth

### PR DESCRIPTION
**Pull request by MichaelYochpaz**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-9715
fixes: https://jira-dc.paloaltonetworks.com/browse/XSUP-33090

## Description
Update the integration to no longer require an API token when using OAuth authentication.

### CODE REVIEW

File: Packs/Okta/Integrations/Okta_v2/Okta_v2_description.md [hunk 1]
```diff
@@ -2,16 +2,15 @@
 Okta API tokens are used to authenticate requests to Okta APIs. 
 
 ### Prerequisites
-1. Sign in to your Okta organization as a user with administrator privileges.
-2. In the Admin Console, select **Security** > **API** from the menu and then select the **Tokens** tab.
+1. Sign in to your Okta organization as a user **with administrator privileges**.
+2. On the **Admin Console**, select **Security** > **API** from the menu, and then select the **Tokens** tab.
```
Comment (ShirleyDenkberg) [inline Packs/Okta/Integrations/Okta_v2/Okta_v2_description.md:6]:
```suggestion
2. In the Admin Console, select **Security** > **API** from the menu, and then select the **Tokens** tab.
```

File: Packs/Okta/Integrations/Okta_v2/Okta_v2_description.md [hunk 2]
```diff
@@ -33,16 +32,22 @@ The following scopes are required for the Okta v2 integration to work properly:
...
-9. In the **Add Public Key** dialog box, click **Generate new key**, and make sure to keep the private key (in PEM format) in somewhere safe.
-10. On the app configuration page, under the **Okta API Scopes** tab, make sure that the required scopes mentioned above are granted.
-
-For more information, see the '[Implement OAuth for Okta
-](https://developer.okta.com/docs/guides/implement-oauth-for-okta/main/)' official documentation article.
+1. Sign in to Okta Admin Console.
+2. In the Admin Console, go to **Applications** > **Applications**.
+3. Click **Create App Integration**.
+4. Select **API Services** as the sign-in method, and click **Next**.
+5. Enter the desired name for the created app (e.g. "Cortex XSOAR"), and click **Save**.
```
Comment (ShirleyDenkberg) [inline Packs/Okta/Integrations/Okta_v2/Okta_v2_description.md:39]:
```suggestion
5. Enter the desired name for the created app (e.g., "Cortex XSOAR"), and click **Save**.
```

File: Packs/Okta/Integrations/Okta_v2/Okta_v2_description.md [hunk 3]
```diff
@@ -33,16 +32,22 @@ The following scopes are required for the Okta v2 integration to work properly:
...
+6. On the app configuration page, under the **General** tab and the **Client Credentials** section, select **Public key / Private key** for the **Client authentication** option.
```
Comment (ShirleyDenkberg) [inline Packs/Okta/Integrations/Okta_v2/Okta_v2_description.md:40]:
```suggestion
6. In the app configuration page, under the **General** tab and the **Client Credentials** section, select **Public key / Private key** for the **Client authentication** option.
```

File: Packs/Okta/Integrations/Okta_v2/Okta_v2_description.md [hunk 4]
```diff
@@ -33,16 +32,22 @@ The following scopes are required for the Okta v2 integration to work properly:
...
+7. Under the newly added **PUBLIC KEYS** section, click the **Add Key** button.
```
Comment (ShirleyDenkberg) [inline Packs/Okta/Integrations/Okta_v2/Okta_v2_description.md:41]:
```suggestion
7. Under the newly added **PUBLIC KEYS** section, click **Add Key**.
```

File: Packs/Okta/Integrations/Okta_v2/Okta_v2_description.md [hunk 5]
```diff
@@ -33,16 +32,22 @@ The following scopes are required for the Okta v2 integration to work properly:
...
+8. In the **Add Public Key** dialog box, click **Generate new key**. Make sure to keep the generated private key (in PEM format) somewhere safe, and click **Save**.
```
Comment (ShirleyDenkberg) [inline Packs/Okta/Integrations/Okta_v2/Okta_v2_description.md:42]:
```suggestion
8. In the **Add Public Key** dialog box, click **Generate new key**. Make sure to copy the generated private key (in PEM format) to somewhere safe, and click **Save**.
```

File: Packs/Okta/Integrations/Okta_v2/Okta_v2_description.md [hunk 6]
```diff
@@ -33,16 +32,22 @@ The following scopes are required for the Okta v2 integration to work properly:
...
+9. Under **General Settings** section:
```
Comment (ShirleyDenkberg) [inline Packs/Okta/Integrations/Okta_v2/Okta_v2_description.md:43]:
```suggestion
9. Under the **General Settings** section:
```

File: Packs/Okta/Integrations/Okta_v2/Okta_v2_description.md [hunk 7]
```diff
@@ -33,16 +32,22 @@ The following scopes are required for the Okta v2 integration to work properly:
...
+   1. Next to the **Proof of possession** label, uncheck the **Require Demonstrating Proof of Possession (DPoP) header in token requests** option if it's selected.
+   2. Next to the **Grant type** label, make sure the **Client Credentials** option is selected, and that the **Token Exchange** option is not selected.
+   3. Click **Save**.
+10. Under the **Okta API Scopes** tab, grant the required scopes mentioned above for the app.
+11. Under the **Admin roles** tab:
+    1. click on the **Edit assignments** button.
+    2. On the dropdown list under "Role", select **Super Administrator**.
+    3. click **Save changes** at the top.
```
Comment (ShirleyDenkberg) [inline Packs/Okta/Integrations/Okta_v2/Okta_v2_description.md:51]:
```suggestion
    1. Click **Edit assignments**.
    2. In the dropdown list under "Role", select **Super Administrator**.
    3. Click **Save changes** at the top.
```

File: Packs/Okta/Integrations/Okta_v2/README.md [hunk 8]
```diff
@@ -1,23 +1,18 @@
...
-### Authentication using API Token
-Okta API tokens are used to authenticate requests to Okta APIs. 
+Integration with Okta's cloud-based identity management service.
 
-#### Prerequisites
-1. Sign in to your Okta organization as a user with administrator privileges.
-2. In the Admin Console, select **Security > API** from the menu and then select the **Tokens** tab.
+## Configure Okta v2 on Cortex XSOAR
+### API Token Authentication Prerequisites
+1. Sign in to your Okta organization as a user **with administrator privileges**.
```
Comment (ShirleyDenkberg) [inline Packs/Okta/Integrations/Okta_v2/README.md:5]:
```suggestion
1. Sign in to your Okta organization as a user with administrator privileges.
```

File: Packs/Okta/Integrations/Okta_v2/README.md [hunk 9]
```diff
@@ -32,21 +27,27 @@ The following scopes are required for the Okta v2 integration to work properly:
...
-6. Enter a name for your app integration.
-7. On the app configuration page, under the **General** tab and the **Client Credentials** section, select **Public key / Private key** for the **Client authentication** option.
-8. Under the newly added **PUBLIC KEYS** section, click the **Add Key** button.
-9. In the **Add Public Key** dialog box, click **Generate new key**, and make sure to keep the private key (in PEM format) in somewhere safe.
-10. On the app configuration page, under the **Okta API Scopes** tab, make sure that the required scopes mentioned above are granted.
+1. Sign in to Okta Admin Console.
+2. In the Admin Console, go to **Applications** > **Applications**.
+3. Click **Create App Integration**.
+4. Select **API Services** as the sign-in method, and click **Next**.
+5. Enter the desired name for the created app (e.g. "Cortex XSOAR"), and click **Save**.
```
Comment (ShirleyDenkberg) [inline Packs/Okta/Integrations/Okta_v2/README.md:34]:
```suggestion
5. Enter the desired name for the created app (e.g., "Cortex XSOAR"), and click **Save**.
```

File: Packs/Okta/Integrations/Okta_v2/README.md [hunk 10]
```diff
@@ -32,21 +27,27 @@ The following scopes are required for the Okta v2 integration to work properly:
...
+6. On the app configuration page, under the **General** tab and the **Client Credentials** section, select **Public key / Private key** for the **Client authentication** option.
```
Comment (ShirleyDenkberg) [inline Packs/Okta/Integrations/Okta_v2/README.md:35]:
```suggestion
6. In the app configuration page, under the **General** tab and the **Client Credentials** section, select **Public key / Private key** for the **Client authentication** option.
```

File: Packs/Okta/Integrations/Okta_v2/README.md [hunk 11]
```diff
@@ -32,21 +27,27 @@ The following scopes are required for the Okta v2 integration to work properly:
...
+7. Under the newly added **PUBLIC KEYS** section, click the **Add Key** button.
```
Comment (ShirleyDenkberg) [inline Packs/Okta/Integrations/Okta_v2/README.md:36]:
```suggestion
7. Under the newly added **PUBLIC KEYS** section, click **Add Key**.
```

File: Packs/Okta/Integrations/Okta_v2/README.md [hunk 12]
```diff
@@ -32,21 +27,27 @@ The following scopes are required for the Okta v2 integration to work properly:
...
+8. In the **Add Public Key** dialog box, click **Generate new key**. Make sure to keep the generated private key (in PEM format) somewhere safe, and click **Save**.
```
Comment (ShirleyDenkberg) [inline Packs/Okta/Integrations/Okta_v2/README.md:37]:
```suggestion
8. In the **Add Public Key** dialog box, click **Generate new key**. Make sure to copy the generated private key (in PEM format) to somewhere safe, and click **Save**.
```

File: Packs/Okta/Integrations/Okta_v2/README.md [hunk 13]
```diff
@@ -32,21 +27,27 @@ The following scopes are required for the Okta v2 integration to work properly:
...
+9. Under **General Settings** section:
```
Comment (ShirleyDenkberg) [inline Packs/Okta/Integrations/Okta_v2/README.md:38]:
```suggestion
9. Under the **General Settings** section:
```

File: Packs/Okta/Integrations/Okta_v2/README.md [hunk 14]
```diff
@@ -32,21 +27,27 @@ The following scopes are required for the Okta v2 integration to work properly:
...
+   1. Next to the **Proof of possession** label, uncheck the **Require Demonstrating Proof of Possession (DPoP) header in token requests** option if it's selected.
+   2. Next to the **Grant type** label, make sure the **Client Credentials** option is selected, and that the **Token Exchange** option is not selected.
+   3. Click **Save**.
+10. Under the **Okta API Scopes** tab, grant the required scopes mentioned above for the app.
+11. Under the **Admin roles** tab:
+    1. click on the **Edit assignments** button.
+    2. On the dropdown list under "Role", select **Super Administrator**.
+    3. click **Save changes** at the top.
```
Comment (ShirleyDenkberg) [inline Packs/Okta/Integrations/Okta_v2/README.md:46]:
```suggestion
    1. Click **Edit assignments**.
    2. In the dropdown list under "Role", select **Super Administrator**.
    3. Click **Save changes** at the top.
```

File: Packs/Okta/ReleaseNotes/3_2_10.md [hunk 15]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### Okta v2
+
+- ***API Token*** field is no longer required when configuring an instance using OAuth authentication.
```
Comment (ShirleyDenkberg) [inline Packs/Okta/ReleaseNotes/3_2_10.md:6]:
```suggestion
*API Token* field is no longer required when configuring an instance using OAuth authentication.  
```

File: Packs/Okta/ReleaseNotes/3_2_10.md [hunk 16]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### Okta v2
+
+- ***API Token*** field is no longer required when configuring an instance using OAuth authentication.  
+  Since the API token was previously used to set the app with a "Super Administrator" role, this role will be needed to be set manually for newly created apps (see the documentation for more info).
```
Comment (ShirleyDenkberg) [inline Packs/Okta/ReleaseNotes/3_2_10.md:7]: "see the documentation"

What documentation?  Perhaps put a link?

**Comment by ShirleyDenkberg** [comment]
@bziser Doc review completed.

**Review by bziser — APPROVED** [review]

**Pull request merged**